### PR TITLE
Add navigation args to `SavedStateHandle`

### PR DESCRIPTION
**Pull request by rubenquadros**
Navigation arguments can be passed when navigating to a destination. These arguments can be passed down to `SavedStateHandle` which can be accessed directly in the `ViewModel`.

### CODE REVIEW

File: koin-androidx-compose/build.gradle [hunk 1]
```diff
@@ -60,6 +60,7 @@ dependencies {
 
     // Compose
     api "androidx.lifecycle:lifecycle-viewmodel-compose:2.4.1"
+    api "androidx.navigation:navigation-compose:2.5.0"
```
Comment (arnaudgiuliani) [inline koin-androidx-compose/build.gradle:63]: for now, the android stack is using navigation `2.4.x`. Version `2.5.0` can be used for next release `3.3.0`

**Review by arnaudgiuliani — CHANGES_REQUESTED** [review]
See my comment on navigation-compose version

File: koin-androidx-compose/build.gradle (hunk 1 above)
Comment (arnaudgiuliani) [reply koin-androidx-compose/build.gradle:63]: let's pass it for version `3.3.0` as we are introducing `navigation-compose`

Comment (rubenquadros) [reply koin-androidx-compose/build.gradle:63]: Sure, will change it to [2.4.2](https://developer.android.com/jetpack/androidx/releases/navigation#version_242_3)

**Review by arnaudgiuliani — APPROVED** [review]

**Pull request merged**